src/core/hle/mii.h

```diff
@@ -0,0 +1,231 @@
+#include "common/common_types.h"
+
+namespace Mii {
+
+#pragma pack(push, 1)
+// Reference: https://github.com/devkitPro/libctru/blob/master/libctru/include/3ds/mii.h
+struct MiiData {
+    u8 magic; ///< Always 3?
+
+    /// Mii options
+    union {
+        u8 raw;
+
+        BitField<0, 1, u8> allow_copying;   ///< True if copying is allowed
+        BitField<1, 1, u8> is_private_name; ///< Private name?
+        BitField<2, 2, u8> region_lock;     ///< Region lock (0=no lock, 1=JPN, 2=USA, 3=EUR)
+        BitField<4, 2, u8> char_set;        ///< Character set (0=JPN+USA+EUR, 1=CHN, 2=KOR, 3=TWN)
+    } mii_options;
+
+    /// Mii position in Mii selector or Mii maker
+    union {
+        u8 raw;
+
+        BitField<0, 4, u8> page_index; ///< Page index of Mii
+        BitField<4, 4, u8> slot_index; ///< Slot offset of Mii on its Page
+    } mii_pos;
+
+    /// Console Identity
+    union {
+        u8 raw;
+
+        BitField<0, 4, u8> unknown0; ///< Mabye padding (always seems to be 0)?
+        BitField<4, 3, u8>
+            origin_console; ///< Console that the Mii was created on (1=WII, 2=DSI, 3=3DS)
+    } console_identity;
+
+    u64_be system_id;      ///< Identifies the system that the Mii was created on (Determines pants)
+    u32_be mii_id;         ///< ID of Mii
+    std::array<u8, 6> mac; ///< Creator's system's full MAC address
+    u16 pad;               ///< Padding
+
+    /// Mii details
+    union {
+        u16_be raw;
+
+        BitField<0, 1, u16> sex;          ///< Sex of Mii (False=Male, True=Female)
+        BitField<1, 4, u16> bday_month;   ///< Month of Mii's birthday
+        BitField<5, 5, u16> bday_day;     ///< Day of Mii's birthday
+        BitField<10, 4, u16> shirt_color; ///< Color of Mii's shirt
+        BitField<14, 1, u16> favorite;    ///< Whether the Mii is one of your 10 favorite Mii's
+    } mii_details;
+
+    std::array<u16_le, 10> mii_name; ///< Name of Mii (Encoded using UTF16)
+    u8 height;                       ///< How tall the Mii is
+    u8 width;                        ///< How wide the Mii is
+
+    /// Face style
+    union {
+        u8 raw;
+
+        BitField<0, 1, u8> disable_sharing; ///< Whether or not Sharing of the Mii is allowed
+        BitField<1, 4, u8> shape;           ///< Face shape
+        BitField<5, 3, u8> skin_color;      ///< Color of skin
+    } face_style;
+
+    /// Face details
+    union {
+        u8 raw;
+
+        BitField<0, 4, u8> wrinkles;
+        BitField<4, 4, u8> makeup;
+    } face_details;
+
+    u8 hair_style;
+
+    /// Hair details
+    union {
+        u8 raw;
+
+        BitField<0, 3, u8> color;
+        BitField<3, 1, u8> flip;
+    } hair_details;
+
+    /// Eye details
+    union {
+        u32_be raw;
+
+        BitField<0, 6, u32> style;
+        BitField<6, 3, u32> color;
+        BitField<9, 4, u32> scale;
+        BitField<13, 3, u32> yscale;
+        BitField<16, 5, u32> rotation;
+        BitField<21, 4, u32> xspacing;
```

Comment:
```suggestion
        BitField<21, 4, u32> x_spacing;
```
same with other similar variables